lib/Dialect/FIRRTL/Transforms/InferResets.cpp

```diff
@@ -1311,40 +1318,79 @@ InferResetsPass::collectAnnos(FModuleOp module) {
-    Value arg = module.getArgument(argNum);
-    if (anno.isClass(fullAsyncResetAnnoClass)) {
-      if (!isa<AsyncResetType>(arg.getType())) {
-        mlir::emitError(arg.getLoc(), "'FullAsyncResetAnnotation' must "
-                                      "target async reset, but targets ")
+  // Helper for checking annotations and determining the reset
+  auto checkAnnotations = [&](Annotation anno, Value arg) {
+    if (anno.isClass(fullResetAnnoClass)) {
+      ResetKind expectedResetKind;
+      if (auto rt = anno.getMember<StringAttr>("resetType")) {
+        if (rt == "sync") {
+          expectedResetKind = ResetKind::Sync;
+        } else if (rt == "async") {
+          expectedResetKind = ResetKind::Async;
+        } else {
+          mlir::emitError(arg.getLoc(),
+                          "'FullResetAnnotation' requires resetType == 'sync' "
+                          "| 'async', but got resetType == ")
+              << rt;
+          anyFailed = true;
+          return true;
+        }
+      } else {
+        mlir::emitError(arg.getLoc(),
+                        "'FullResetAnnotation' requires resetType == "
+                        "'sync' | 'async', but got no resetType");
+        anyFailed = true;
+        return true;
+      }
+      auto badType = false;
+      if (expectedResetKind == ResetKind::Async &&
+          !isa<AsyncResetType>(arg.getType())) {
+        badType = true;
+      } else if (expectedResetKind == ResetKind::Sync) {
+        if (!isa<UIntType>(arg.getType()) ||
+            cast<UIntType>(arg.getType()).getWidth() != 1) {
+          badType = true;
+        }
+      }
+      if (badType) {
+        auto kind = resetKindToStringRef(expectedResetKind);
+        mlir::emitError(arg.getLoc(),
+                        "'FullResetAnnotation' with resetType == '")
+            << kind << "' must target " << kind << " reset, but targets "
             << arg.getType();
         anyFailed = true;
         return true;
       }
+
       reset = arg;
       conflictingAnnos.insert({anno, reset.getLoc()});
 
       return false;
     }
-    if (anno.isClass(ignoreFullAsyncResetAnnoClass)) {
+    if (anno.isClass(excludeFromFullResetAnnoClass)) {
       anyFailed = true;
       mlir::emitError(arg.getLoc(),
-                      "'IgnoreFullAsyncResetAnnotation' cannot target port; "
-                      "must target module instead");
+                      "'ExcludeFromFullResetAnnotation' cannot "
+                      "target port/wire/node; must target module instead");
       return true;
     }
     return false;
-  });
+  };
+
+  AnnotationSet::removePortAnnotations(module,
+                                       [&](unsigned argNum, Annotation anno) {
+                                         Value arg = module.getArgument(argNum);
+                                         return checkAnnotations(anno, arg);
+                                       });
```

Comment:
clang-format did me dirty here, unclear why this is tabbed so weird whereas the very similar `AnnotationSet::removeAnnotations(op, [&](Annotation anno) {` below is tabbed more reasonably 🤷‍♂️

Reply:
The unlucky situation where you ever-so-slightly surpass 80 characters ://